### PR TITLE
Use g_timeout instead of gtk_timeout. Don't release already released source

### LINKED ISSUE #19: Remove use of deprecated gtk_timeout_add etc
gtk_timeout_add() and related has allegedly been deprecated since GTK 2.4.

Looks like the replacement is g_timeout_add() but I have not researched it fully.

### PR DESCRIPTION
**Pull request by oskarb**
In some of our timer callbacks, we return FALSE/G_SOURCE_REMOVE to trigger autoremoval of the source. In those cases, we must also remove the source id, so we don't try to remove it again. It may have been reused, in which case we would remove something unintended.

This fixes the runtime warnings telling us about this issue.

### CODE REVIEW

File: src/idle-timer.c [hunk 1]
```diff
@@ -281,7 +281,7 @@ notice_events_timer (gpointer closure)
   XSetErrorHandler (old_handler);
 
   /* we pop once, and we don't pop again */
-  return 0;
+  return G_SOURCE_REMOVE;
```
Comment (markuspg) [inline src/idle-timer.c]: Could you please change the return type of this function to `gboolean`? Then we are fully in line with `GSourceFunc` signature (not that it makes any difference in effect, but it avoids confusion).

Comment (markuspg) [reply src/idle-timer.c]: I guess `check_pointer_timer` and `idle_timeout_main_loop` are candidates for the same treatment.

File: src/journal.c [hunk 2]
```diff
@@ -712,13 +712,13 @@ on_close_clicked_cb (GtkWidget *w, gpointer data)
 	 * pop after the destroy has happened. */
 	if (wig->hover_timeout_id)
 	{
-		gtk_timeout_remove (wig->hover_timeout_id);
+		g_source_remove (wig->hover_timeout_id);
```
Comment (markuspg) [inline src/journal.c]: In this file the return types of `hover_kill_func` and `hover_timer_func` could be adjusted too.

**Review by markuspg — DISMISSED** [review]

**Comment by markuspg** [comment]
Apart of a few very minor comments (which concern code predating this pull request) everything looks good, thank you :-)

File: src/idle-timer.c (hunk 1 above)
Comment (oskarb) [reply src/idle-timer.c]: Sounds good. I will wait for the reformatting to land first.

Comment (markuspg) [reply src/idle-timer.c]: You should be good to go :-)

**Comment by oskarb** [comment]
@markuspg Needs reapproval.

**Review by markuspg — APPROVED** [review]
Looks good, thank you :-)

**Pull request merged**